### DISCUSSION TITLE
Ideas: Legacy volume control

### DISCUSSION THREAD

**Discussion by Anix-x** [valinet/ExplorerPatcher#125] [Ideas]
What about implementing a registry hack for the Win7-style volume control?

https://winaero.com/enable-old-volume-control-in-windows-10/

I think it is not going to be removed any time soon because it is part of sound mixer.

**Comment by valinet** [comment]
Slowly but surely, we reimplement the whole Winaero Tweaker in here... idk, yeah, I guess... the problem is, the "Taskbar" section is already getting too large. Idk what to say...

**Reply by Anix-x** [reply]
Maybe, a separate "tray" section?

**Reply by valinet** [reply]
Which would include, beside that?

**Reply by valinet** [reply]
All the tray related settings?

**Reply by valinet** [reply]
Yeah, idk, the file is here: https://github.com/valinet/ExplorerPatcher/blob/master/ExplorerPatcher/settings.reg. You can draft a proposal for reorganization based off that. I can make the GUI load a file if it finds one besides it. That way, you can test easily. The format is based off registry files, with some auxiliary comments of mine to help the code build the UI.

**Reply by Anix-x** [reply]
* Tray pop-up menus (center etc)
* Flyout behavor
* Network and sharing center
* Hide Control Center button
etc...

Possibly, in the future, "Hide the show desktop button". "Legacy time and calendar", etc.

**Reply by valinet** [reply]
Why in the future? Those can already be supported...

**Reply by valinet** [reply]
As I said, start from that file, make changes to it. I will upload a version soon that will load the settings from the folder the DLL is in, if it finds a file there. That way, it can be used rapidly for debugging.

**Reply by valinet** [reply]
@Anix-x Okay, latest version (30.4) allows loading the `settings.reg` file from the same folder where the DLL is located. Thus, download the file from GitHub, put it in a folder with the DLL, and then launch the DLL from cmd like so: `rundll32 dxgi.dll,ZZGUI`. It will load the UI from the `reg` file. Any changes you make to the `reg` file, reload the DLL using `rundll32` and inspect them. Thus, you can easily play with the interface however you want, add more settings etc.
